src/libraries/System.Private.CoreLib/src/System/Threading/SynchronizationContext.cs

```diff
@@ -19,7 +21,7 @@ public SynchronizationContext()
 
         public virtual void Send(SendOrPostCallback d, object? state) => d(state);
 
-        public virtual void Post(SendOrPostCallback d, object? state) => ThreadPool.QueueUserWorkItem(static s => s.d(s.state), (d, state), preferLocal: false);
+        public virtual void Post(SendOrPostCallback d, object? state) => ThreadPool.QueueUserWorkItem(static s => s.Key(s.Value), new KeyValuePair<SendOrPostCallback, object?>(d, state), preferLocal: false);
```

Comment:
This is odd use of KeyValuePair. There is no key-value relationship between the two items.

Reply:
I agree but `s.d(s.state)` is not descriptive either.